internal/libooniengine/engine.h

```diff
@@ -19,7 +26,53 @@ char *OONIEngineVersion(void);
 /// OONIEngineFreeMemory frees the memory allocated by the engine.
 ///
 /// @param ptr a void pointer refering to the memory to be freed.
-void OONIENgineFreeMemory(void *ptr);
+void OONIEngineFreeMemory(void *ptr);
+
+/// OONIEngineCall starts a new OONITask using the given [req].
+///
+/// @param req A JSON string, owned by the caller, that
+/// contains the configuration for the task to start.  
+///
+/// @return Zero on failure, nonzero on success. If the return value
+/// is nonzero, a task is running. In such a case, the caller is
+/// responsible to eventually dispose of the task using OONIEngineFreeMemory.
+OONITask OONIEngineCall(char *req);
+
+/// OONIEngineWaitForNextEvent awaits on the [task] event queue until
+/// a new event is available or the given [timeout] expires.
+///
+/// @param task Task handle returned by OONIEngineCall.
+///
+/// @param timeout Timeout in milliseconds. If the timeout is zero
+/// or negative, this function would potentially block forever.
+///
+/// @return A NULL pointer on failure, non-NULL otherwise. If the return
+/// value is non-NULL, the caller takes ownership of the OONIMessage
+/// pointer and MUST free it using OONIMessageFree when done using it.
+///
+/// This function will return NULL:
+///
+/// 1. when the timeout expires;
+///
+/// 2. if [task] is done;
+///
+/// 3. if [task] is zero or does not refer to a valid task;
+///
+/// 4. if we cannot protobuf serialize the message;
+///
+/// 5. possibly because of other unknown internal errors.
+///
+/// In short, you cannot reliably determine whether a task is done by
+/// checking whether this function has returned NULL.
```

Comment:
Did the original API include a function for checking whether a task is done?

Reply:
Yes, thank you for pointing this out (I missed it). I have implemented this in future commits (along with the go code)